SimpleExec/Command.cs

```diff
@@ -19,14 +20,14 @@ public static void Run(string name, string args, string workingDirectory, bool n
             }
         }
 
-        public static async Task RunAsync(string name, string args, string workingDirectory, bool noEcho)
+        public static async Task RunAsync(string name, string args, string workingDirectory, bool noEcho, CancellationToken cancellationToken)
         {
             using (var process = new Process())
             {
                 process.StartInfo = ProcessStartInfo.Create(name, args, workingDirectory, false);
-                await process.RunAsync(noEcho).ConfigureAwait(false);
+                await process.RunAsync(noEcho, cancellationToken).ConfigureAwait(false);
 
-                if (process.ExitCode != 0)
+                if (process.ExitCode != 0 && !cancellationToken.IsCancellationRequested)

```

Comment:
If `cancellationToken` is `null`, this will throw a `NullReferenceException`, and unless I'm mistaken, that will be the case when the `RunAsync(string name, string args, string workingDirectory, bool noEcho)` overload is called, since that passes `default` for the `cancellationToken` parameter.

Reply:
`CancellationToken` is a struct type, so we shouldn't get a `NullReferenceException` here. The properties and methods around `CancellationToken` are designed as such that a `default(CancellationToken)` won't execute things, i.e., the token isn't considered canceled, the cancellation registration just returns out, etc.

Reply:
Aha! Yes, I completely missed that. All OK then.